cloud/azure/mysql/variables.tf

```diff
@@ -275,3 +275,53 @@ variable "replication_lag_threshold_major" {
   type        = number
   default     = 100
 }
+
+# failed_connections detector
+
+variable "failed_connections_notifications" {
+  description = "Notification recipients list per severity overridden for failed_connections detector"
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "failed_connections_aggregation_function" {
+  description = "Aggregation function and group by for failed_connections detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ""
+}
+
+variable "failed_connections_transformation_function" {
+  description = "Transformation function for failed_connections detector (i.e. \".mean(over='5m')\")"
+  type        = string
+  default     = ""
+}
+
+variable "failed_connections_disabled" {
+  description = "Disable all alerting rules for failed_connections detector"
+  type        = bool
+  default     = null
+}
+
+variable "failed_connections_disabled_major" {
+  description = "Disable major alerting rule for failed_connections detector"
+  type        = bool
+  default     = null
+}
+
+variable "failed_connections_disabled_critical" {
+  description = "Disable critical alerting rule for failed_connections detector"
+  type        = bool
+  default     = null
+}
+
+variable "failed_connections_threshold_major" {
+  description = "Major threshold for failed_connections detector"
+  type        = number
+  default     = 10
+}
+
+variable "failed_connections_threshold_critical" {
+  description = "Critical threshold for failed_connections detector"
+  type        = number
+  default     = 20
```

Comment:
not sure about this value, maybe it may be better to compare with `active_connections` metric?

Reply:
it depends on what this metric is and sadly the azure doc is not very useful as usual: https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported#microsoftdbformysqlservers

in my knowledge of mysql monitoring world I only know two metrics (from mysql point of view) which could match:
- [aborted clients](https://dev.mysql.com/doc/refman/5.6/en/server-status-variables.html#statvar_Aborted_clients)
- [aborted connects](https://dev.mysql.com/doc/refman/5.6/en/server-status-variables.html#statvar_Aborted_connects)

I doubt Microsoft collects more advanced data to create this metric but it could be a sum of both :man_shrugging: 

assuming no and it simply is the popular `aborted_connects` value so I agree with @Shr3ps it seems not relevant to alert on a static threshold here considering the number of "normal" errors it could happen: https://dev.mysql.com/doc/refman/5.6/en/communication-errors.html. 

I suppose wrong password could be a security problem of course but for now our modules are not security focused and if we add security oriented alerts I think it should be in separate modules than the "basic" infra alerts purpose.

on the [agent based version of this module](https://github.com/claranet/terraform-signalfx-detectors/tree/master/database/mysql) I provide a way in readme to collect this metric but I don't use it in detectors because I think the only relevant error to alert is the max_conections reached scenario which is better monitored "before" the error happen.

Here you don't have the max_connections value and as I understand https://docs.microsoft.com/en-us/azure/mysql/concepts-limits#server-parameters it could be tuned so we cannot trust the instance type or default value config so it seems impossible to calculate the % of connections over max capacity.

Without that it makes sens to use this metric even if it could contains not relevant errors.
Using the active connection as suggested by @Shr3ps could mitigate this risk but I don't think make a simple percentage is a good idea. it will probably never raise alert, too late or false.

Of course with more advanced work like double "AND" conditions to correlate:
- the number of failed connections value > 0
- with a "% of change" of active connections number ~=0% (i.e. `.rateofchange().abs()`)
we can suppose if errors number increases in the same time the active connections number stagnate this is probably a max connection reached scenario.

may be third condition to compare the value of active connections from the present to a little `timeshift` like 15m will help to be sure it is a valid alert but it could lead to auto resolve after a long period

Else I only see workarounds:
- to simply force the user to set the threshold depending on the max connections limit and make a detector (in separate module than this one because will be used "per" database). in this case you alert on max connection reached and you do not need this metric anymore.
- to use https://docs.signalfx.com/en/latest/detect-alert/alert-condition-reference/hist-anomaly.html instead of threshold based detector on failed or even number of active connections. it will not really alert on max connections reached (as for your current detector) but should be smarter and avoid "stupid" alerts.